packages/react/src/components/MultiSelect/FilterableMultiSelect.tsx

```diff
@@ -502,8 +686,32 @@ const FilterableMultiSelect = React.forwardRef(function FilterableMultiSelect(
       )}
     />
   );
-});
+}
+
+type PropsWithRef<P, T> = PropsWithoutRef<P> & RefAttributes<T>;
```

Comment:
So… here be dragons… My reasoning is explained in the code because future generations might reasonably get perplexed & possibly even face-slappingly angry without some explation for such fugliness...

This is, however, at least to my knowledge, one of those shining examples where typescript is just the best bad thing we have.

Reply:
Thanks for commenting it so well. @mbarrer or @jdharvey-ibm would either of you have thoughts on situations like this?

Reply:
Some food for thought when looking at this from a user's perspective. Here's the error you get when using the `FilterableMultiSelect` component:

<img width="1101" alt="image" src="https://user-images.githubusercontent.com/51208233/234699768-00ae81a6-d7f0-4708-8d89-a7eae7f0ec6c.png">

Compare that to using `FilterableMultiSelectRenderFunction` directly:

<img width="1115" alt="image" src="https://user-images.githubusercontent.com/51208233/234699916-b78ab4ae-5829-4d93-ae3c-138d98c6a9bb.png">

This isn't a deal breaker, but it is definitely a more confusing error in the first image than the second.

It looks like this might also introduce an additional component render into the tree, which could be a minor debugging and performance hit. I didn't verify this, but the:
```tsx
const Comp = forwardRef<HTMLDivElement, Props>(
  FilterableMultiSelectRenderFunction
);
return <Comp {...props} />;
```
makes me think that might be the case.

I feel like we had a similar discussion to this for the `ComboBox` before I went on leave at the end of last year. Were there any directional decisions made on this type of thing in that PR?

Reply:
Yeah, the type printing is a mess!

Honestly, I'd rather forgo the `forwardRef` wrapper entirely and just offer a new prop in lieu of `rootRef` / `inputRef` / …

This, in my view, needlessly complicates things on many levels… some of which are the generics / extra wrappers / differences in exports